cpp/src/neighbors/ivf_flat/ivf_flat_build.cuh

```diff
@@ -118,10 +119,10 @@ auto clone(const raft::resources& res, const index<T, IdxT>& source) -> index<T,
  * @param veclen size of vectorized loads/stores; must satisfy `dim % veclen == 0`.
  *
  */
-template <typename T, typename IdxT, typename LabelT, bool gather_src = false>
+template <typename T, typename IdxT, typename LabelT, bool gather_src = false, typename SourceIdxT>
```

Comment:
As `SourceIdxT` is meant to be inferred, we should put it before template arguments with default params.

```suggestion
template <typename T, typename IdxT, typename LabelT, typename SourceIdxT, bool gather_src = false>
```